vita3k/https/src/https.cpp

```diff
@@ -236,285 +63,94 @@ std::string get_web_regex_result(const std::string &url, const std::regex &regex
-        LOG_ERROR("404 Not Found");
-        close_ssl(https.ssl);
-        close_socket(https.sockfd);
-
-        return false;
-    }
-
-    // Check if bytes read is diferent of header size
-    if (bytes_read < (header_size * 0.99f)) {
-        LOG_ERROR("Error reading header: {}/{}\n{}", bytes_read, header_size, read_buffer.data());
-        close_ssl(https.ssl);
-        close_socket(https.sockfd);
-        return false;
-    }
-
-    // Create lambda to get current time in milliseconds
-    const auto get_current_time_ms = []() {
-        return std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
-    };
-
-    // Open the output file
-    std::ofstream outfile(output_file_path, std::ios::app | std::ios::binary);
-
-    // Set the initial downloaded file size to calculate the remaining time of download
-    auto initial_downloaded_file_size = downloaded_file_size;
-
-    // Set the initial time to calculate the remaining time of download
-    auto start_time = get_current_time_ms();
-
-    float progress_percent = 0.f;
-    uint64_t remaining_time = 0;
-    ProgressState progress_state{};
-
-    while (progress_state.download && bytes_read) {
-        if (!progress_state.pause) {
-            if ((bytes_read = SSL_read(https.ssl, read_buffer.data(), static_cast<uint32_t>(read_buffer.size())))) {
-                // Write the read buffer to the output file
-                outfile.write(read_buffer.data(), bytes_read);
-
-                // Update the downloaded file size
-                downloaded_file_size += bytes_read;
-
-                if (progress_callback) {
-                    // Update progress percent
-                    progress_percent = static_cast<float>(downloaded_file_size) / static_cast<float>(file_size) * 100.0f;
-
-                    // Calculate elapsed time since start of download in seconds
-                    const auto elapsed_time_ms = std::difftime(get_current_time_ms(), start_time);
-
-                    // Calculate remaining time in seconds
-                    const auto downloaded_bytes = static_cast<double>(downloaded_file_size - initial_downloaded_file_size);
-                    const auto remaining_bytes = static_cast<double>(file_size - downloaded_file_size);
-                    remaining_time = static_cast<uint64_t>((remaining_bytes / downloaded_bytes) * elapsed_time_ms) / 1000;
-                }
+        while (callback_result->pause) {
+            // Check if the download has been aborted
+            if (!callback_result->download) {
+                return 1; // Returning anything thats not 0 aborts the request
             }
-        } else {
-            // Sleep for 100ms to not consume CPU
-            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        };
 
-            // Reset initial downloaded file size and start time to calculate remaining time correctly when resume download
```

Comment:
Your code no works here, you need keep my code when is on pause for update time and init file size
If not time raimaining is juste broken